spec/std/string/grapheme_spec.cr

```diff
@@ -19,6 +20,20 @@ describe String::Grapheme do
     end.should eq "f"
   end
 
+  describe "#each_char" do
+    it_iterates "string", ['f', 'o', 'o'], String::Grapheme.new("foo").each_char
+    it_iterates "string", ['🙂', '🙂'], String::Grapheme.new("🙂🙂").each_char
```

Comment:
Perhaps a bit obsessive, but IMO it's missing two cases: an empty string, and (a) grapheme(s) with more than one codepoint.

Reply:
A grapheme consisting of an empty string should be impossible to acquire. I don't think it's necessary to test that. I would see that as an invalid state with undefined behaviour.
These two highlighted examples have more than one codepoint. Maybe it's a bit confusing that they are not actual graphemes. I chose sequences of codepoints that don't form an actual grapheme for demonstration purposes. This is unrelated to the grapheme breaker algorithm, so it doesn't matter. Let me know if you think I should use actual graphemes instead.